src/baldr/graphreader.cc

```diff
@@ -357,9 +358,20 @@ GraphReader::GraphReader(const boost::property_tree::ptree& pt,
   // validate tile url
   if (!tile_url_.empty() && tile_url_.find(GraphTile::kTilePathPattern) == std::string::npos)
     throw std::runtime_error("Not found tilePath pattern in tile url");
+
   // Reserve cache (based on whether using individual tile files or shared,
   // mmap'd file
   cache_->Reserve(tile_extract_->tiles.empty() ? AVERAGE_TILE_SIZE : AVERAGE_MM_TILE_SIZE);
+
+  // Initialize the incident cache singleton if we have any kind of configuration to do so
+  // if the configuration is wrong or any kind of problem occurs this throws
+  enable_incidents_ = !pt.get<std::string>("incident_log", "").empty() ||
+                      !pt.get<std::string>("incident_dir", "").empty();
+  if (enable_incidents_) {
```

Comment:
we enable incidents and have the singleton load whatever incidents are there based on config. if its not in the config its completey disabled. the call below will spawn a single background thread which is responsible for loading incidents continually

Reply:
This would make a pretty helpful code comment?

Reply:
most of it is in the comment above, do you specifically mean the part about the background thread?